### PR TITLE
TCP Concepts Guide created as companion piece to TCP CLIs

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

File: site/en/reference/cli/concepts/tcp.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Test TCP Functionality With OT CLI
+
+Thread networks offer several `cli tcp` commands for testing peer-to-peer communication
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/tcp.md:3]: I didn't notice this in the other UDP review... I don't think "Thread networks offer several `cli tcp` commands", I think OpenThread offers tcp commands for use with a Thread network.  Can you reword to reflect that?  (And in the UDP guide too to match).

File: site/en/reference/cli/concepts/tcp.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Based on the example steps shown above, the following output would be expected:
+
+1. After Node 2 runs the `tcp connect` command, Node 2 should receive:
+   the message `TCP: Connection established`.
+1. Node 1 should then receive the messages (with example IPv6 address and port):
+    * `Accepted connection from [fe80:0:0:0:8f3:f602:bf9b:52f2]:49152`
+    * `TCP: Connection established`
+1. After Node 2 runs the `tcp send` command, Node 1 should receive:
+   the message: `TCP: Received 5 bytes: hello`
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/tcp.md:119]: These bullets don't have consistent formatting...

probably don't need the colon after "receive" in 1 and 3, then there's a colon after message in 3

File: site/en/reference/cli/concepts/tcp.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Use the `tcp init` command to initialize the TCP module to begin TCP communication.
+The TCP module can then perform many functions, such as listening for incoming
+connections using the example TCP listener provided by the `tcp` CLI.
+If you want to deinitialize the example TCP listener and the example TCP endpoint,
+you can issue the `tcp deinit` command.
+
+### `bind` command
+
+After you initialize the example TCP endpoint, you can run a `tcp bind`
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/tcp.md:42]: Can you rewrite these to not use "you can".

Like this one, dropping "you can" would be fine.

**Review by Vyrastas — CHANGES_REQUESTED** [review]

**Review by abtink — APPROVED** [review]
LGTM.

**Review by Vyrastas — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**